daemon/daemon.go

```diff
@@ -922,12 +922,17 @@ func (daemon *Daemon) GetRemappedUIDGID() (int, int) {
 	return uid, gid
 }
 
-// tempDir returns the default directory to use for temporary files.
-func tempDir(rootDir string, rootUID, rootGID int) (string, error) {
+// prepareTempDir prepares and returns the default directory to use
+// for temporary files.
+// If it doesn't exist, it is created. If it exists, its content is removed.
+func prepareTempDir(rootDir string, rootUID, rootGID int) (string, error) {
 	var tmpDir string
 	if tmpDir = os.Getenv("DOCKER_TMPDIR"); tmpDir == "" {
 		tmpDir = filepath.Join(rootDir, "tmp")
+		os.RemoveAll(tmpDir)
```

Comment:
:ok_hand: for waiting on D4M (ping @justincormack ?) to have a look. Although, it's a temp directory, nothing is there should be assumed to survive be valid after a restart ;-)

Regarding the second point, I'm not sure I understand. If you remove the directory, you also remove its content.